enhancements/authentication/configuring-webhook-token-authenticators.md

```diff
@@ -0,0 +1,176 @@
+   server by using the `webhookTokenAuthneticators` field instead of the current patching of the
+   kube-apiserver.
+
+### Non-Goals
+
+1. provide a partial authentication operator to any external authenticator.  Once the standard oauth-server
+   is disabled, authentication configuration is *entirely* the responsibility of the new authentication stack.
+2. allow both the built-in oauth-server and another authentication stack to be enabled at the same time.
+
+## Proposal
+
+### Implementation
+
+#### API
+
+`authentications.config.openshift.io/v1` already contains the `spec.webhookTokenAuthenticators`. It
+expects an array of references to secrets in the `openshift-config` namespace. The field is currently
+unused so it is safe to leverage it for the purposes of configuring webhook token authenticators.
+
+Any secret reference which gets set to this field has to be validated. In 4.4, it will only be
+possible to configure a single element for the `wehbhookTokenAuthenticators` field and the referenced
+kubeconfigs will only be allowed to use the `-data` forms of their fields to configure
+certificate authority, client certificate and key.
+
+This path is compatible with kubernetes as it is possible to directly map the referenced
+secrets' contents to files and specify these in the kube-apiserver configuration.
+
+#### Backend - kube-apiserver-operator
+
+- observes the changes in the `authentication/cluster` resource and performs validation of any references
+  in `webhookTokenAuthenticators` field
+- it pushes any webhook authenticator that passes the validation to `openshift-kube-apiserver/webhooktokenauthenticators`
+  secret, where `key=authenticator_name` where `authenticator_name` is either the name of the referenced secret or
+  the CR
+- if no webhook authenticators pass validation, the operator keeps the previous state
+- redeploys the kube-apiserver static pods while specifying the new
+  mounted files in the `webhookTokenAuthenticators` [config field](https://github.com/openshift/api/blob/992bc3a41fe6c4e5bcf27dc8080f9e5c114c33fc/kubecontrolplane/v1/types.go#L146).
+
+#### Backend - Integrated OAuth Server Configuration
+
+It's cluster-authentication-operator's responsibility to:
+- observe `authentication.config/cluster` field
+- if set to "IntegratedOAuth", it creates secret `openshift-config/integrated-oauth` and adds it to
+  the `webhookTokenAuthenticators` field of the `authentication.config/cluster` resource
+- the above secret contains details that point kube-apiserver to the oauth-apiserver for token reviews
+
+To perform token reviews for tokens issued by the integrated oauth-server, the new oauth-apiserver needs to
+be able to handle token review requests, such capability is currently missing and needs to be implemented.
+
+When responding after a successful authentication, the oauth-apiserver needs to return all of the users
```

Comment:
nit: user's ?

nit: Is `group` equivalent to `role`, as in RBAC, or is something different and oath specific? If the former, using role exclusively might be clearer.

Reply:
Here, group is an object that can hold multiple users. See `oc explain group`